### ISSUE TITLE
Use ARC Kubernetes mode to run benchmark

### ISSUE THREAD

**Issue by xuzhao9** [pytorch/benchmark#2357]
ARC Kubernetes mode has the highest security level among all runner modes.
We can keep using CPU-only instance for docker building, but all GPU instances should use ARC Kubernetes mode.

Reference: https://www.stepsecurity.io/blog/how-to-use-docker-in-actions-runner-controller-runners-securelly

**Comment by kit1980** [comment]
On PyTorch, we decided to stop ARC migration.
Is this still relevant for TorchBench?

**Comment by xuzhao9** [comment]
Yes, it is still relevant for TorchBench, TB uses ARC runners on GCP K8s cluster for PR and nightly testing: https://github.com/pytorch/benchmark/tree/main/docker/infra

If we migrate all OSS workflows to Internal, or if pytorch CI provides capacity for GCP A100 runners, we can also deprecate the ARC runner setup.
